framework/include/multiapps/MultiApp.h

```diff
@@ -40,6 +40,39 @@ class NumericVector;
 template <>
 InputParameters validParams<MultiApp>();
 
+/// Holds app partitioning information relevant to the a particular rank for a
+/// multiapp scenario.
+struct LocalRankConfig
+{
+  /// The number of apps that should/will be run locally on this rank.
+  unsigned int num_local_apps;
+  /// The (global) index of the first local app for this rank.  All ranks that
+  /// are used to perform multi-proc parallel runs for a given app will have the
+  /// same first_local_app_index as each other.
+  unsigned int first_local_app_index;
+  /// This is true if this rank is the primary/zero rank for a (sub)app slot.
+  /// A slot is all ranks that are grouped together to run a single (sub)app
+  /// together.  This field will be true for exactly one rank in each slot.
+  /// This is important for things like multiapp transfers where you want to
+  /// only transfer data to a given subapp once even though it may be running on
+  /// multiple procs/ranks.
+  bool am_first_local_rank;
```

Comment:
Can we use "is", am seems wrong to me for some arbitrary reason.
```suggestion
  bool is_first_local_rank;
```

Reply:
done